test/metabase/test/data/drill.clj

```diff
@@ -0,0 +1,187 @@
+                             (if (= base-type :type/DateTime)
+                               (format "CAST(to_timestamp(`%s`, 'YYYY-MM-dd''T''HH:mm:ss.SSSZ') AS TIMESTAMP) AS `%s`"
+                                       field-name field-name)
+                               (format "CAST(`%s` AS %s) AS `%s`"
+                                       field-name
+                                       (field-base-type->sql-type base-type)
+                                       field-name))))
+                      (interpose ", ")
+                      (apply str))
+                 ", CAST(`id` as INTEGER) AS `id` ")
+            table-path)))
+
+(defn drop-table-if-exists-sql [{:keys [database-name]} {:keys [table-name]}]
+  (format "DROP VIEW IF EXISTS %s" (str "dfs.tmp.`" database-name "_" table-name "`")))
+
+;; workaround for DRILL-5136 (can't use prepared statements for DDL)
+(defn drill-execute-sql! [driver context dbdef sql]
+  (let [sql (some-> sql s/trim)]
+    (when (and (seq sql)
+               ;; make sure SQL isn't just semicolons
+               (not (s/blank? (s/replace sql #";" ""))))
+      ;; Remove excess semicolons, otherwise snippy DBs like Oracle will barf
+      (let [sql (s/replace sql #";+" ";")]
+        (try
+          (with-open [conn (jdbc/get-connection (generic/database->spec driver context dbdef))]
+            (with-open [statement (.createStatement conn)]
+              (.execute statement sql)))
+          (catch SQLException e
+            (printf "Caught SQLException:\n%s\n"
+                    (with-out-str (jdbc/print-sql-exception-chain e)))
+            (throw e))
+          (catch Throwable e
+            (printf "Caught Exception: %s %s\n%s\n" (class e) (.getMessage e)
+                    (with-out-str (.printStackTrace e)))
+            (throw e)))))))
+
+(defn sequentially-execute-sql!
+  "Alternative implementation of `execute-sql!` that executes statements one at a time for drivers
+   that don't support executing multiple statements at once.
+
+   Since there are some cases were you might want to execute compound statements without splitting, an upside-down ampersand (`⅋`) is understood as an
+   \"escaped\" semicolon in the resulting SQL statement."
+  [driver context dbdef sql]
+  (when sql
+    (try
+      (with-open [conn (jdbc/get-connection (generic/database->spec driver context dbdef))]
+        (doseq [statement (map s/trim (s/split (s/replace sql #"⅋" ";") #";+"))]
+          (when (seq statement)
+            (with-open [sql-statement (.createStatement conn)]
+              (.execute sql-statement statement)))))
+      (catch SQLException e
+        (printf "Caught SQLException:\n%s\n"
+                (with-out-str (jdbc/print-sql-exception-chain e)))
+        (throw e))
+      (catch Throwable e
+        (printf "Caught Exception: %s %s\n%s\n" (class e) (.getMessage e)
+                (with-out-str (.printStackTrace e)))
+        (throw e)))))
```

Comment:
Why is this copied over from the generic-sql namespace? Can't you just use that implementation?

Reply:
No, this is different, since it does not use prepared statements (Drill does not support prepared statements).

Reply:
Wouldn't it be possible to write an implementation something like this?

```clojure
(defn sequentially-execute-sql! [driver context dbdef sql]
  (generic-sql/sequentially-execute-sql! driver context dbdef (unprepare sql))
```

(Just call the normal function with an unprepared version of the SQL)

I'm not comfortable with copying entire sections of code to change one little thing. It makes this PR way bigger than it needs to be and if we're not careful about keeping as much code shared as possible the driver code with spiral out of control in terms of complexity

Reply:
Kind of. I renamed it to `sequentially-execute-sql-without-prepared-statements!` and managed to reuse some code from generic-sql, but there's still a little bit of repetition in `execute-without-prepared-statement!` that I wasn't able to get rid of. I hope it's good enough.